src/plugins/discover/public/application/main/utils/get_switch_index_pattern_app_state.ts

```diff
@@ -28,19 +28,30 @@ export function getSwitchIndexPatternAppState(
       )
     : currentColumns;
   const columns = nextColumns.length ? nextColumns : [];
+
   // when switching from an index pattern with timeField to an index pattern without timeField
   // filter out sorting by timeField in case it is set. index patterns without timeField don't
   // prepend this field in the table, so in legacy grid you would need to add this column to
   // remove sorting
-  const nextSort = getSortArray(currentSort, nextIndexPattern).filter((value) => {
+  let nextSort = getSortArray(currentSort, nextIndexPattern).filter((value) => {
     return nextIndexPattern.timeFieldName || value[0] !== currentIndexPattern.timeFieldName;
   });
+
+  if (nextIndexPattern.isTimeBased() && !nextSort.length) {
+    // set default sorting if it was not set
+    nextSort = [[nextIndexPattern.timeFieldName, sortDirection]];
+  } else if (nextIndexPattern.isTimeBased() && currentIndexPattern.isTimeBased()) {
+    // switch time fields
+    nextSort = nextSort.map((cur) =>
+      cur[0] === currentIndexPattern.timeFieldName
+        ? [nextIndexPattern.timeFieldName, sortDirection]
```

Comment:
`sortDirection` is the default sortDirection. So if the user's sorting is different to the default one, this would change the direction. I think it's better UX to keep the direction the user selected previously for the time field.